### PR TITLE
bugfix:fix type casting problem when using redis as registry 

### LINKED ISSUE #2160: seata-spring-boot-starter 注册时，选用redis注册报错
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

采用redis作为注册器的时候，redis.db 类型转换异常

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:
redis 作为注册中心时发生的 
```
Just paste your stack trace here!
```
image

### Ⅳ. How to reproduce it (as minimally and precisely as possible)
修改 RegistryRedisProperties 中 db 中类型

### Ⅵ. Environment:

- JDK version : 1.8
- OS : win10
- Others:

### PR DESCRIPTION
**Pull request by guojingyinan219**
…s/2160

### Ⅰ. Describe what this PR did
fix the problem of spring redis register auto injection error ,the issues address: 
 https://github.com/seata/seata/issues/2160 

### Ⅱ. Does this pull request fix one issue?

fixes #2160

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
unit test

### Ⅳ. Describe how to verify it
 spring-redis  register   auto injection

### CODE REVIEW

**Comment by guojingyinan219** [comment]
@renyuanxin111 类型已经修复，fix #2160

**Comment by slievrly** [comment]
@guojingyinan219 Can you adapt other attributes of redis?
image
This pr is also good, other attributes are not optional.

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/registry/RegistryRedisProperties.java [hunk 1]
```diff
@@ -27,7 +27,11 @@
 @ConfigurationProperties(prefix = REGISTRY_REDIS_PREFIX)
 public class RegistryRedisProperties {
     private String serverAddr = "localhost:6379";
-    private String db = "0";
+    private int db = 0;
```
Comment (slievrly) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/registry/RegistryRedisProperties.java]: @guojingyinan219 Can you adapt other attributes of redis?
image
This pr is also good, other attributes are not optional.

**Comment by guojingyinan219** [comment]
@slievrly I don't know how to write this unit test. I package seata-spring-boot-starter and run in my program with debug, it's ok

**Comment by guojingyinan219** [comment]
image
if look like you describe, serverAddr  can not read  and return Object ,but is ok

File: seata-spring-boot-starter/src/main/resources/spring/redis_auto_injectio_test.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:context="http://www.springframework.org/schema/context"
+       xsi:schemaLocation="
+	   http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd
+       http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd">
+
+
+    <bean id="registryRedisProperties"
```
Comment (slievrly) [inline seata-spring-boot-starter/src/main/resources/spring/redis_auto_injectio_test.xml]: @guojingyinan219 remove to test/resource ?

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/registry/RegistryRedisProperties.java [hunk 3]
```diff
@@ -27,7 +27,11 @@
...
+    private String password;
+    private String cluster;
+    private int timeout = 0;
```
Comment (xingfudeshi) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/registry/RegistryRedisProperties.java]: Please add these config items into files under script folder.

Comment (slievrly) [reply seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/registry/RegistryRedisProperties.java]: https://github.com/seata/seata/blob/develop/script/client/conf/registry.conf
https://github.com/seata/seata/tree/develop/script/client/spring

File: seata-spring-boot-starter/src/test/java/io/seata/spring/boot/autoconfigure/RedisAutoInjectionTypeConvertTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import static org.mockito.Mockito.mock;
+
+/**
+ * @version 1.0
+ * @ClassName RedisAutoInjectionTypeConvertTest
+ * @Description
+ * @Author zhangheng
+ * @Date 2020/1/14 17:09
```
Comment (xingfudeshi) [inline seata-spring-boot-starter/src/test/java/io/seata/spring/boot/autoconfigure/RedisAutoInjectionTypeConvertTest.java]: Please remove annotations other than @Author.Thank u.

**Pull request merged**